src/main/java/com/redhat/labs/lodestar/service/GitService.java

```diff
@@ -0,0 +1,337 @@
+
+		Optional<ProjectTreeNode> artifactsTreeNode = projectTree.getArtifactsProjectTreeNode();
+		if (artifactsTreeNode.isEmpty()) {
+			return Optional.empty();
+		}
+
+		// get default branch
+		String branch = findDefaultBranch(projectTree.getProjectId());
+
+		return Optional.ofNullable(
+				gitlabRestClient.getFile(projectTree.getProjectId(), artifactsTreeNode.get().getPath(), branch));
+
+	}
+
+	/*
+	 * Returns a {@link List} of {@link Artifact}s for the given {@link File}. An
+	 * empty {@link List} is returned if the {@link File} is not found or contains
+	 * no {@link Artifact}s.
+	 */
+	List<Artifact> parseFile(Optional<File> file) {
+
+		List<Artifact> artifacts = new ArrayList<>();
+
+		if (file.isEmpty()) {
+			return artifacts;
+		}
+
+		File artifactsFile = file.get();
+
+		artifactsFile.decodeFileAttributes();
+
+		if (ARTIFACT_FILE.equals(artifactsFile.getFilePath())) {
+			artifacts = parseArtifactsFile(artifactsFile);
+		} else if (ENGAGEMENT_FILE.equals(artifactsFile.getFilePath())) {
+			artifacts = parseEngagementFile(artifactsFile);
+		}
+
+		return artifacts;
+
+	}
+
+	/**
+	 * Returns a {@link List} of {@link Artifact}s from the given {@link File}. An
+	 * empty {@link List} is returned if no file found or contains no
+	 * {@link Artifact}s.
+	 * 
+	 * @param file
+	 * @return
+	 */
+	List<Artifact> parseArtifactsFile(File file) {
+
+		if (null == file || null == file.getContent() || file.getContent().isBlank()) {
+			return new ArrayList<>();
+		}
+
+		return Arrays.asList(jsonb.fromJson(file.getContent(), Artifact[].class));
+
+	}
+
+	/**
+	 * Returns a {@link List} of {@link Artifact}s from the given {@link File}. An
+	 * empty {@link List} is returned if no file found or contains no
+	 * {@link Artifact}s.
+	 * 
+	 * @param file
+	 * @return
+	 */
+	List<Artifact> parseEngagementFile(File file) {
+
+		if (null == file || null == file.getContent() || file.getContent().isBlank()) {
+			return new ArrayList<>();
+		}
+
+		Engagement engagement = jsonb.fromJson(file.getContent(), Engagement.class);
+
+		// set engagement uuid on each artifacts
+		List<Artifact> artifacts = null == engagement.getArtifacts() ? new ArrayList<>() : engagement.getArtifacts();
+		artifacts.stream().forEach(a -> a.setEngagementUuid(engagement.getUuid()));
+
+		return artifacts;
+
+	}
+
+	/**
+	 * Returns a {@link List} of {@link Project} found in the configured group or
+	 * subgroups.
+	 * 
+	 * @param includeSubgroups
+	 * @return
+	 */
+	List<Project> getProjectsByGroup(Boolean includeSubgroups) {
+
+		Response response = null;
+		PagedResults<Project> page = new PagedResults<>(pageSize);
+
+		while (page.hasMore()) {
+			response = gitlabRestClient.getProjectsbyGroup(groupParentId, includeSubgroups, pageSize, page.getNumber());
+			page.update(response, new GenericType<List<Project>>() {
+			});
+		}
+
+		if (null != response) {
+			response.close();
+		}
+
+		return page.getResults();
+	}
```

Comment:
Not exactly sure what is happening here but I think I mentioned elsewhere that we should leave project / groups to the git-api and call that service to get the data needed. I'm assuming we will have to update the git-api to get the needed data with some sort of efficiency

Reply:
now calling git-api.